support-files/mariadb.service

```diff
@@ -0,0 +1,104 @@
+#
+#  /etc/systemd/system/mariadb.service
+#
+
+[Unit]
+Description=MariaDB database server
+After=network.target
+After=syslog.target
+Requires=%p.socket
+
+[Install]
+WantedBy=multi-user.target
+Alias=mysql.service
+Alias=mysqld.service
+
+
+[Service]
+
+##############################################################################
+## Core requirements
+##
+
+Type=Notify
+
+# Setting this to true can break replication and the Type=Notify settings
+PrivateNetwork=false
+
+##############################################################################
+## Package maintainers
+##
+
+User=mysql
+
+# Execute pre and post scripts as root
+PermissionsStartOnly=true
+
+# Needed to create system tables etc.
+ExecStartPre=/usr/bin/mariadb-systemd-start
+
+# Start main service
+# skip-networking as its socket activation based
+ExecStart=/usr/sbin/mysqld $OPTIONS
+
+KillMode=process
+KillSignal=SIGTERM
+# Don't want to see an automated SIGKILL ever
+SendSIGKILL=no
+
+RestartPreventExitStatus=1
+Restart=on-fail
+RestartSec=5s
+
+PrivateTmp=true
+PrivateDevices=true
+
+UMask=077
+WorkingDirectory=/
```

Comment:
this is mostly from https://mariadb.atlassian.net/browse/MDEV-427?focusedCommentId=59314&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-59314

Anything in particular you object/need explanation to/of?